File: frontend/types/weather.ts
export interface GeoLocation {
  name: string;
  lat: number;
  lon: number;
  country: string;
  state?: string;
}

export interface WeatherCondition {
  id: number;
  main: string;
  description: string;
  icon: string;
}

export interface CurrentWeather {
  city: string;
  country: string;
  lat: number;
  lon: number;
  temp: number;
  feels_like: number;
  temp_min: number;
  temp_max: number;
  humidity: number;
  pressure: number;
  wind_speed: number;
  wind_deg: number;
  visibility: number;
  weather: WeatherCondition[];
  dt: number;
  sunrise: number;
  sunset: number;
  uv_index?: number;
  aqi?: number;
}

export interface HourlyItem {
  dt: number;
  temp: number;
  weather: WeatherCondition[];
  rain?: number;
  pop: number;
}

export interface ForecastItem {
  dt: number;
  temp: number;
  temp_min: number;
  temp_max: number;
  humidity: number;
  weather: WeatherCondition[];
  wind_speed: number;
  pop: number;
  rain?: number;
}

export interface Forecast {
  city: string;
  country: string;
  items: ForecastItem[];
  hourly: HourlyItem[];
}

File: backend/app/models/weather.py
from typing import Optional, List
from pydantic import BaseModel


class GeoLocation(BaseModel):
    name: str
    lat: float
    lon: float
    country: str
    state: Optional[str] = None


class WeatherCondition(BaseModel):
    id: int
    main: str
    description: str
    icon: str


class CurrentWeather(BaseModel):
    city: str
    country: str
    lat: float
    lon: float
    temp: float
    feels_like: float
    temp_min: float
    temp_max: float
    humidity: int
    pressure: int
    wind_speed: float
    wind_deg: int
    visibility: int
    weather: List[WeatherCondition]
    dt: int
    sunrise: int
    sunset: int
    uv_index: Optional[float] = None
    aqi: Optional[int] = None


class HourlyItem(BaseModel):
    dt: int
    temp: float
    weather: List[WeatherCondition]
    rain: Optional[float] = None
    pop: float = 0


class ForecastItem(BaseModel):
    dt: int
    temp: float
    temp_min: float
    temp_max: float
    humidity: int
    weather: List[WeatherCondition]
    wind_speed: float
    pop: float
    rain: Optional[float] = None


class Forecast(BaseModel):
    city: str
    country: str
    items: List[ForecastItem]
    hourly: List[HourlyItem]
